src/unstract/sdk/file_storage/fs_impl.py

```diff
@@ -51,6 +52,8 @@ def read(
                 if seek_position > 0:
                     file_handle.seek(seek_position)
                 return file_handle.read(length)
+        except FileNotFoundError as e:
+            raise e
```

Comment:
@gaya3-zipstack [MINOR] We can log the exception trace here.

Reply:
Will add it the next PR @harini-venkataraman

Reply:
@gaya3-zipstack I hope this is on track somewhere, hope it doesn't get missed.